### PR TITLE
chore: add test for tx priority

### LINKED ISSUE #513: testing: Add integration test for tx prioritization
When we upgraded to cosmos-sdk v0.46, we gain the ability prioritized txs in the mempool. Since this feature is so critical, we need to test this in an integration test.

### PR DESCRIPTION
**Pull request by evan-forbes**
## Overview

adds a simple integration test for tx priority in the mempool

closes #513 #2377

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

**Review by rootulp — APPROVED** [review]

**Comment by rootulp** [comment]
Maybe also closes https://github.com/celestiaorg/celestia-app/issues/2377

**Review by staheri14 — DISMISSED** [review]
LGTM! added some suggestions.

File: app/test/priority_test.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+		hashes = append(hashes, resp.TxHash)
+	}
+
+	err := s.cctx.WaitForNextBlock()
+	require.NoError(t, err)
+
+	// get the responses for each tx for analysis and sort by height
+	// note: use rpc types because they contain the tx index
+	heightMap := make(map[int64][]*rpctypes.ResultTx)
+	for _, hash := range hashes {
```
[the remark this answers is not in the document]
Comment (staheri14) [reply app/test/priority_test.go:96]: [update] `BroadcastTx` method does not wait for tx commitment in a block, that's why we need to wait for their inclusion in the block by invocation of `ConfirmTx`.  Anyway, if we could add the `mode` parameter to the `BroadCastTx`,  and pass `BroadcastMode_BROADCAST_MODE_BLOCK` we could avoid another call to `ConfirmTx`.  Not directly relevant to this PR though.

File: app/test/priority_test.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+	for _, acc := range cfg.Accounts {
+		addr := testfactory.GetAddress(s.cctx.Keyring, acc)
+		signer, err := user.SetupSigner(s.cctx.GoContext(), s.cctx.Keyring, s.cctx.GRPCClient, addr, s.ecfg)
+		signer.SetPollTime(time.Millisecond * 300)
+		require.NoError(t, err)
+		s.signers = append(s.signers, signer)
+	}
+}
+
+func (s *PriorityTestSuite) TestPriorityByGasPrice() {
```
[the remark this answers is not in the document]
Comment (evan-forbes) [reply app/test/priority_test.go:70]: :+1:  https://github.com/celestiaorg/celestia-app/pull/2394/commits/b5a11dcc40a05dbcde42279437c6cc1c2238e27e

File: pkg/user/tx_options.go [hunk 3]
```diff
@@ -66,6 +66,18 @@ func SetFeeGranter(feeGranter sdk.AccAddress) TxOption {
 	}
 }
 
+func SetGasPrice(gasLimit uint64, gasPrice float64) TxOption {
```
[the remark this answers is not in the document]
Comment (evan-forbes) [reply pkg/user/tx_options.go]: :+1: https://github.com/celestiaorg/celestia-app/pull/2394/commits/05db952cb2472b28b4fe97c340901973ae01c175

**Review by rootulp — APPROVED** [review]

**Pull request merged**